api/main.go

```diff
@@ -77,545 +76,707 @@ func NewApiServer(logger *zap.SugaredLogger, address string, metaAddr string, sr
+			loggerWithResource.Errorw("Failed to read client request", "Error", err)
 			return err
 		}
-		proxyStream, err := serv.meta.GetUsers(stream.Context())
+		proxyStream, err := serv.meta.GetUsers(ctx)
 		if err != nil {
+			loggerWithResource.Errorw("Failed to get users from server", "Error", err)
 			return err
 		}
-		sErr := proxyStream.Send(name)
+		sErr := proxyStream.Send(nameRequest)
 		if sErr != nil {
+			loggerWithResource.Errorw("Failed to send user to the server", "user", nameRequest.Name, "Error", err)
 			return sErr
 		}
 		res, err := proxyStream.Recv()
 		if err != nil {
+			loggerWithResource.Errorw("Failed to receive users from server", "user", nameRequest.Name, "Error", err)
 			return err
 		}
 		sendErr := stream.Send(res)
 		if sendErr != nil {
+			loggerWithResource.Errorw("Failed to send users to client", "user", nameRequest.Name, "Error", err)
 			return sendErr
 		}
 	}
 }
 
 func (serv *MetadataServer) GetFeatures(stream pb.Api_GetFeaturesServer) error {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
 	for {
-		name, err := stream.Recv()
+		nameRequest, err := stream.Recv()
+		loggerWithResource := logger.WithResource("feature", nameRequest.Name.Name, "")
+		loggerWithResource.Debugw("Get feature %v from stream", nameRequest.Name)
+		nameRequest.RequestId = requestID
 		if err == io.EOF {
 			return nil
 		}
 		if err != nil {
-			serv.Logger.Errorf("Failed to read client request: %v", err)
+			loggerWithResource.Errorw("Failed to read client request", "Error", err)
 			return err
 		}
-		proxyStream, err := serv.meta.GetFeatures(stream.Context())
+		proxyStream, err := serv.meta.GetFeatures(ctx)
 		if err != nil {
+			loggerWithResource.Errorw("Failed to get features from server", "feature", nameRequest.Name, "Error", err)
 			return err
 		}
-		sErr := proxyStream.Send(name)
+		sErr := proxyStream.Send(nameRequest)
 		if sErr != nil {
+			loggerWithResource.Errorw("Failed to send feature to the server", "feature", nameRequest.Name, "Error", err)
 			return sErr
 		}
 		res, err := proxyStream.Recv()
 		if err != nil {
+			loggerWithResource.Errorw("Failed to receive features from server", "feature", nameRequest.Name, "Error", err)
 			return err
 		}
 		sendErr := stream.Send(res)
 		if sendErr != nil {
+			loggerWithResource.Errorw("Failed to send features to client", "feature", nameRequest.Name, "Error", err)
 			return sendErr
 		}
 	}
 }
 
 func (serv *MetadataServer) GetFeatureVariants(stream pb.Api_GetFeatureVariantsServer) error {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(stream.Context())
 	for {
-		nameVariant, err := stream.Recv()
+		nameVariantRequest, err := stream.Recv()
+		loggerWithResource := logger.WithResource("feature", nameVariantRequest.NameVariant.Name, nameVariantRequest.NameVariant.Variant)
+		loggerWithResource.Debugw("Get feature variant %v from stream", nameVariantRequest.NameVariant)
```

Comment:
this This could be an info, this is a bug since the w doesn't have the key. Also no need to add these since its in the With